### PR TITLE
[capi] added unittest for get layer

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1:     [neuralnet] enhance print function to print graph info
    
     - Enhance print function in neuralnet to print layer connection
       printing input tensor info is omitted.
     - Enhance print function in layer_node to print layer properties
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 2:     [capi] added unittest for get layer
    
     - Added 4 unittest for capi ml_train_model_get_layer
     - get layer api is validate using get summary function
       but this should be replace by layer getProperty(#1875)
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 1]
```diff
@@ -544,6 +544,89 @@ TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
...
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_layer_set_property(add_layer, "name=inputlayer", NULL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_add_layer(model, add_layer);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_get_layer(model, "inputlayer", &get_layer);
+  EXPECT_EQ(status, ML_ERROR_NONE);
```
Comment (myungjoo) [inline test/tizen_capi/unittest_tizen_capi.cpp:570]: Can you test the validity of the get_layer?

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 2]
```diff
@@ -544,6 +544,89 @@ TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
...
+
+  status = ml_train_model_destroy(model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Neural Network Model Get Layer Test
+ */
+TEST(nntrainer_capi_nnmodel, getLayer_03_n) {
```
Comment (myungjoo) [inline test/tizen_capi/unittest_tizen_capi.cpp:604]: For 2 new positive cases, you need 2 or more negative cases

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 1 above)
Comment (jijoongmoon) [reply test/tizen_capi/unittest_tizen_capi.cpp:570]: One of the way is that you could set the input dim of input layer when you create and check the dim after you call the get_layer.

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 3]
```diff
@@ -544,6 +544,89 @@ TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
...
+  status = ml_train_model_add_layer(model, add_layer);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_get_layer(model, "unknwon", &get_layer);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_model_destroy(model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
```
Comment (jijoongmoon) [inline test/tizen_capi/unittest_tizen_capi.cpp:629]: I also recommend a negative test case for the call of the get layer function after the compilation or initialization of the model.

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 2 above)
Comment (lhs8928) [reply test/tizen_capi/unittest_tizen_capi.cpp:633]: Done

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 3 above)
Comment (lhs8928) [reply test/tizen_capi/unittest_tizen_capi.cpp:658]: Done

File: test/tizen_capi/unittest_tizen_capi.cpp (hunk 1 above)
Comment (lhs8928) [reply test/tizen_capi/unittest_tizen_capi.cpp:571]: I check it by set name of layer which is access by get layer API.

As @jijoongmoon mention in issue #1875 this should be replace by getProperty API which will be implement.

File: nntrainer/models/neuralnet.cpp [hunk 4]
```diff
@@ -877,6 +878,10 @@ int NeuralNetwork::setDataBuffer(const DatasetModeType &mode,
 
 int NeuralNetwork::getLayer(const char *name,
                             std::shared_ptr<ml::train::Layer> *layer) {
+  if (compiled) {
+    return ML_ERROR_NOT_SUPPORTED;
+  }
```
Comment (myungjoo) [inline nntrainer/models/neuralnet.cpp:883]: ```compiled```? ```!compiled```?

Comment (lhs8928) [reply nntrainer/models/neuralnet.cpp:883]: It's compiled.
Once the model is compiled then the graph structure is fixed.
So attempts to modifying the graph structure eg) changing input shape or input layer is prohibithed if model is compiled

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 5]
```diff
@@ -544,6 +545,131 @@ TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
...
+ * @brief Neural Network Model Get Layer Test
+ */
+TEST(nntrainer_capi_nnmodel, getLayer_02_p) {
+  int status = ML_ERROR_NONE;
+
+  ml_train_model_h model;
+  ml_train_layer_h get_layer;
+
+  std::string default_name = "inputlayer", modified_name = "renamed_inputlayer";
+  char *default_summary, *modified_summary = nullptr;
```
Comment (songgot) [inline test/tizen_capi/unittest_tizen_capi.cpp:587]: default_summary and modified_summary seem to need free after use.

Comment (lhs8928) [reply test/tizen_capi/unittest_tizen_capi.cpp:587]: Done

File: test/tizen_capi/unittest_tizen_capi.cpp [hunk 6]
```diff
@@ -544,6 +545,133 @@ TEST(nntrainer_capi_nnmodel, addLayer_05_n) {
...
+  ScopedIni s("getLayer_04_n", {model_base, inputlayer});
+
+  status = ml_train_model_construct_with_conf(s.getIniName().c_str(), &model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_compile(model, NULL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_get_layer(model, "inputlayer", &get_layer);
+  EXPECT_EQ(status, ML_ERROR_NOT_SUPPORTED);
```
[the remark this answers is not in the document]
Comment (songgot) [reply test/tizen_capi/unittest_tizen_capi.cpp:669]: I misunderstood. thank you.

**Review by songgot — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**